src/api/procedures/createGroup.ts

```diff
@@ -80,22 +81,40 @@ export async function prepareCreateGroup(
   const rawTicker = stringToTicker(ticker, context);
   const { transactions } = permissionsLikeToPermissions(permissions, context);
 
-  const { custom } = await token.permissions.getGroups();
+  const { custom, known } = await token.permissions.getGroups();
+  const allGroups = [...custom, ...known];
 
-  const currentGroupPermissions = await P.map(custom, group => group.getPermissions());
+  const currentGroupPermissions = await P.map(allGroups, group => group.getPermissions());
 
-  if (
-    currentGroupPermissions.some(({ transactions: transactionPermissions }) => {
+  let groupId: PermissionGroupType | BigNumber | undefined;
+
+  const duplicatedGroups = currentGroupPermissions.some(
+    ({ transactions: transactionPermissions }, i) => {
       const sortedTransactions = transactions && {
         ...transactions,
         values: [...transactions.values].sort(),
       };
-      return isEqual(transactionPermissions, sortedTransactions);
-    })
-  ) {
+      const result = isEqual(transactionPermissions, sortedTransactions);
+
+      if (result) {
+        const permissionGroup = allGroups[i];
+        groupId =
+          permissionGroup instanceof CustomPermissionGroup
+            ? permissionGroup.id
+            : permissionGroup.type;
+      }
```

Comment:
You shouldn't use `some` for side effects. You should use `find` to see if there is an existing group, throw the error and use the found group's ID/type for the data